--- a/core/rfb.js	2025-10-02 22:18:27.444440990 -0400
+++ b/core/rfb.js	2025-10-02 23:17:29.338761880 -0400
@@ -159,6 +159,10 @@
         this._clipboardText = null;
         this._clipboardServerCapabilitiesActions = {};
         this._clipboardServerCapabilitiesFormats = {};
+        this._clipboardAutoSync = "denied";
+        this._clipboardAutoSyncWritePerm = null;
+        this._clipboardAutoSyncReadPerm = null;
+        this._initClipboardAutoSyncState();
 
         // Internal objects
         this._sock = null;              // Websock object
@@ -210,6 +214,7 @@
             handleGesture: this._handleGesture.bind(this),
             handleRSAAESCredentialsRequired: this._handleRSAAESCredentialsRequired.bind(this),
             handleRSAAESServerVerification: this._handleRSAAESServerVerification.bind(this),
+            handleClipboardFocus: this._handleClipboardFocus.bind(this),
         };
 
         // main setup
@@ -346,6 +351,8 @@
         this._updateClip();
     }
 
+    get clipboardAutoSync() { return this._clipboardAutoSync; }
+
     get scaleViewport() { return this._scaleViewport; }
     set scaleViewport(scale) {
         this._scaleViewport = scale;
@@ -606,6 +613,9 @@
         this._canvas.addEventListener("gesturemove", this._eventHandlers.handleGesture);
         this._canvas.addEventListener("gestureend", this._eventHandlers.handleGesture);
 
+        // Focus event for automatic clipboard synchronization
+        this._canvas.addEventListener("focus", this._eventHandlers.handleClipboardFocus);
+
         Log.Debug("<< RFB.connect");
     }
 
@@ -623,6 +633,7 @@
         this._canvas.removeEventListener('contextmenu', this._eventHandlers.handleMouse);
         this._canvas.removeEventListener("mousedown", this._eventHandlers.focusCanvas);
         this._canvas.removeEventListener("touchstart", this._eventHandlers.focusCanvas);
+        this._canvas.removeEventListener("focus", this._eventHandlers.handleClipboardFocus);
         this._resizeObserver.disconnect();
         this._keyboard.ungrab();
         this._gestures.detach();
@@ -2327,6 +2338,109 @@
         return this._fail("Unexpected SetColorMapEntries message");
     }
 
+    async _initClipboardAutoSyncState() {
+        if (!window.isSecureContext) {
+            Log.Info("clipboard auto sync requires a secure context");
+            return;
+        } else if (!(navigator?.permissions?.query)) {
+            Log.Info("clipboard auto sync requires support of Permissions API");
+            return;
+        } else if (!(navigator?.clipboard?.writeText) || !(navigator?.clipboard?.readText)) {
+            Log.Info("clipboard auto sync requires support of Clipboard API");
+            return;
+        }
+
+        try {
+            const writePerm = await navigator.permissions.query(
+                {name: "clipboard-write", allowWithoutGesture: false});
+            const readPerm = await navigator.permissions.query(
+                {name: "clipboard-read", allowWithoutGesture: false});
+
+            Log.Info(`clipboard-write permission status is ${writePerm.state}`);
+            Log.Info(`clipboard-read permission status is ${readPerm.state}`);
+
+            writePerm.onchange = () => {
+                Log.Info(`clipboard-write permission status changed to ${writePerm.state}`);
+                this._updateClipboardAutoSyncState();
+            }
+            readPerm.onchange = () => {
+                Log.Info(`clipboard-read permission status changed to ${readPerm.state}.`);
+                this._updateClipboardAutoSyncState();
+            }
+
+            this._clipboardAutoSyncWritePerm = writePerm;
+            this._clipboardAutoSyncReadPerm = readPerm;
+            this._updateClipboardAutoSyncState();
+        } catch (error) {
+            if (error instanceof TypeError) {
+                Log.Error("clipboard permissions query failed: ", error);
+            } else {
+                Log.Info("clipboard permission not supported");
+            }
+        }
+    }
+
+    _updateClipboardAutoSyncState() {
+        let autoSync = "denied";
+
+        if (this._clipboardAutoSyncWritePerm.state === "denied" ||
+            this._clipboardAutoSyncReadPerm.state === "denied") {
+            autoSync = "denied";
+        } else if (this._clipboardAutoSyncWritePerm.state === "granted" &&
+            this._clipboardAutoSyncReadPerm.state === "granted") {
+            autoSync = "granted";
+        } else if (this._clipboardAutoSyncWritePerm.state === "prompt" ||
+            this._clipboardAutoSyncReadPerm.state === "prompt") {
+            autoSync = "prompt";
+        }
+
+        if (autoSync != this._clipboardAutoSync) {
+            Log.Info(`clipboard auto sync changed to ${autoSync}`)
+            this._clipboardAutoSync = autoSync;
+            this.dispatchEvent(new CustomEvent(
+                "clipboardautosync",
+                { detail: { state: autoSync } }));
+        }
+    }
+
+    async _handleClipboardFocus() {
+        if (this._viewOnly) return;
+        if (this._clipboardAutoSync === "denied") return;
+
+        // In "prompt" state, reading the clipboard will prompt
+        // the user for permission.
+
+        try {
+            const text = await navigator.clipboard.readText();
+            this.clipboardPasteFrom(text);
+        } catch (error) {
+            if (error.name === "NotAllowedError") {
+                // User did not grant permission.
+                Log.Info("clipboard read failed: ", error);
+            } else {
+                Log.Error("clipboard read failed: ", error);
+            }
+        }
+    }
+
+    async _writeClipboard(text) {
+        if (this._viewOnly) return;
+        if (this._clipboardAutoSync === "granted") {
+            (async () => {
+                try {
+                    await navigator.clipboard.writeText(text);
+                } catch (error) {
+                    Log.Error("clipboard write failed: ", error);
+                }
+            })();
+        } else {
+            // Dispatch event for the alternative clipboard
+            this.dispatchEvent(
+                new CustomEvent("clipboard", {detail: {text: text}})
+            );
+        }
+    }
+
     _handleServerCutText() {
         Log.Debug("ServerCutText");
 
@@ -2346,10 +2460,7 @@
                 return true;
             }
 
-            this.dispatchEvent(new CustomEvent(
-                "clipboard",
-                { detail: { text: text } }));
-
+            this._writeClipboard(text);
         } else {
             //Extended msg.
             length = Math.abs(length);
@@ -2484,9 +2595,7 @@
 
                     textData = textData.replaceAll("\r\n", "\n");
 
-                    this.dispatchEvent(new CustomEvent(
-                        "clipboard",
-                        { detail: { text: textData } }));
+                    this._writeClipboard(textData);
                 }
             } else {
                 return this._fail("Unexpected action in extended clipboard message: " + actions);
