diff --git a/src/main.c b/src/main.c
index 41fc47e..2b40fa1 100644
--- a/src/main.c
+++ b/src/main.c
@@ -44,6 +44,10 @@
 
 #include <errno.h>
 
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
 Display *display = NULL;
 Window rootWindow;
 int rootWidth, rootHeight;
@@ -97,6 +101,8 @@ static void SendExit(void);
 static void SendReload(void);
 static void SendJWMMessage(const char *message);
 
+static void SavePID();
+
 static char *displayString = NULL;
 
 char *configPath = NULL;
@@ -191,6 +197,9 @@ int main(int argc, char *argv[])
       /* Start up the JWM components. */
       Startup();
 
+      // JWM is now ready.
+      SavePID();
+
       /* The main event loop. */
       EventLoop();
 
@@ -670,3 +679,25 @@ void SendJWMMessage(const char *message)
    CloseConnection();
 }
 
+void SavePID()
+{
+   int fd = open("/var/run/jwm/jwm.pid", O_CREAT|O_WRONLY|O_TRUNC);
+   if (fd < 0) {
+      Warning(_("pid file open failed: %s (%d)"), strerror(errno), errno);
+      return;
+   }
+
+   char buf[128];
+   snprintf(buf, sizeof(buf), "%ld", (long)getpid());
+
+   int rc = write(fd, buf, strlen(buf));
+   if (rc < 0) {
+      Warning(_("pid file write failed: %s (%d)"), strerror(errno), errno);
+      close(fd);
+      unlink("/var/run/jwm/jwm.pid");
+      return;
+   }
+
+   fchmod(fd, S_IRUSR|S_IWUSR|S_IRGRP|S_IROTH);
+   close(fd);
+}
