index 924a4d4aa..e1205d2d6 100644
--- a/src/daemon/main.c
+++ b/src/daemon/main.c
@@ -555,7 +555,7 @@ int main(int argc, char *argv[]) {
     pa_log_set_level(PA_LOG_NOTICE);
     pa_log_set_flags(PA_LOG_COLORS|PA_LOG_PRINT_FILE|PA_LOG_PRINT_LEVEL, PA_LOG_RESET);
 
-#if !defined(HAVE_BIND_NOW) && defined(__linux__) && defined(__OPTIMIZE__)
+#if 0 //!defined(HAVE_BIND_NOW) && defined(__linux__) && defined(__OPTIMIZE__)
     /*
        Disable lazy relocations to make usage of external libraries
        more deterministic for our RT threads. We abuse __OPTIMIZE__ as
