--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -341,7 +341,7 @@ if (ENABLE_NETTLE)
 endif()
 
 # Check for PAM library
-if(UNIX AND NOT APPLE)
+if(ENABLE_UNIX_PASSWORD_VALIDATOR)
   find_package(PAM REQUIRED)
 endif()
 
--- a/common/rfb/CMakeLists.txt
+++ b/common/rfb/CMakeLists.txt
@@ -76,7 +76,7 @@ if(WIN32)
   target_sources(rfb PRIVATE WinPasswdValidator.cxx)
 endif(WIN32)
 
-if(UNIX AND NOT APPLE)
+if(ENABLE_UNIX_PASSWORD_VALIDATOR)
   target_sources(rfb PRIVATE UnixPasswordValidator.cxx)
   target_include_directories(rfb SYSTEM PRIVATE ${PAM_INCLUDE_DIRS})
   target_link_libraries(rfb ${PAM_LIBRARIES})
--- a/common/rfb/SSecurityPlain.cxx
+++ b/common/rfb/SSecurityPlain.cxx
@@ -28,7 +28,7 @@
 #include <rfb/SConnection.h>
 #include <rfb/Exception.h>
 #include <rdr/InStream.h>
-#if !defined(WIN32) && !defined(__APPLE__)
+#if(ENABLE_UNIX_PASSWORD_VALIDATOR)
 #include <rfb/UnixPasswordValidator.h>
 #include <unistd.h>
 #include <pwd.h>
@@ -53,7 +53,7 @@ bool PasswordValidator::validUser(const char* username)
   for (const char* user : plainUsers) {
     if (strcmp(user, "*") == 0)
       return true;
-#if !defined(WIN32) && !defined(__APPLE__)
+#if defined(ENABLE_UNIX_PASSWORD_VALIDATOR)
     if (strcmp(user, "%u") == 0) {
       struct passwd *pw = getpwnam(username);
       if (pw && pw->pw_uid == getuid())
@@ -72,7 +72,7 @@ SSecurityPlain::SSecurityPlain(SConnection* sc_) : SSecurity(sc_)
 {
 #ifdef WIN32
   valid = new WinPasswdValidator();
-#elif !defined(__APPLE__)
+#elif(ENABLE_UNIX_PASSWORD_VALIDATOR)
   valid = new UnixPasswordValidator();
 #else
   valid = nullptr;
--- a/common/rfb/SSecurityRSAAES.cxx
+++ b/common/rfb/SSecurityRSAAES.cxx
@@ -47,7 +47,7 @@
 #include <rfb/SSecurityRSAAES.h>
 #include <rfb/SConnection.h>
 #include <rfb/Exception.h>
-#if !defined(WIN32) && !defined(__APPLE__)
+#if defined(ENABLE_UNIX_PASSWORD_VALIDATOR)
 #include <rfb/UnixPasswordValidator.h>
 #endif
 #ifdef WIN32
@@ -577,10 +577,10 @@ bool SSecurityRSAAES::readCredentials()
 
 void SSecurityRSAAES::verifyUserPass()
 {
-#ifndef __APPLE__
+#if defined(WIN32) || defined(ENABLE_UNIX_PASSWORD_VALIDATOR)
 #ifdef WIN32
   WinPasswdValidator* valid = new WinPasswdValidator();
-#elif !defined(__APPLE__)
+#elif defined(ENABLE_UNIX_PASSWORD_VALIDATOR)
   UnixPasswordValidator *valid = new UnixPasswordValidator();
 #endif
   std::string msg = "Authentication failed";
--- a/unix/xserver/hw/vnc/RFBGlue.cc
+++ b/unix/xserver/hw/vnc/RFBGlue.cc
@@ -32,7 +32,9 @@
 
 #include <network/TcpSocket.h>
 
+#if defined(ENABLE_UNIX_PASSWORD_VALIDATOR)
 #include <rfb/UnixPasswordValidator.h>
+#endif
 
 #include "RFBGlue.h"
 
@@ -241,5 +243,7 @@ void vncSetDisplayName(const char *displayNumStr)
 {
   std::string displayName(":");
   displayName += displayNumStr;
+#if defined(ENABLE_UNIX_PASSWORD_VALIDATOR)
   rfb::UnixPasswordValidator::setDisplayName(displayName);
+#endif
 }
